test/functional/tests/lazy_writes/recovery/test_recovery_unplug.py

```diff
@@ -0,0 +1,172 @@
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+@pytest.mark.parametrize("cache_mode", CacheMode.with_traits(CacheModeTrait.LazyWrites))
+@pytest.mark.parametrize("cls", [CacheLineSize.LINE_4KiB, CacheLineSize.LINE_64KiB])
+@pytest.mark.require_plugin("power_control")
+def test_recovery_unplug_cache_raw(cache_mode, cls):
+    """
+            title: Test for recovery after cache drive removal - test on raw device.
+            description: |
+              Verify that unflushed data can be safely recovered after, when SSD drive is removed
+              after write completion - test on raw device.
+            pass_criteria:
+              - CAS recovers successfully after cache drive unplug
+              - No data corruption
+    """
+    with TestRun.step("Prepare devices"):
+        cache_disk = TestRun.disks['cache']
+        core_disk = TestRun.disks['core']
+        cache_disk.create_partitions([Size(2, Unit.GibiByte)])
+        core_disk.create_partitions([Size(16, Unit.GibiByte)])
+        cache_device = cache_disk.partitions[0]
+        core_device = core_disk.partitions[0]
+
+    with TestRun.step("Create test files."):
+        source_file, target_file = create_test_files(test_file_size)
+
+    with TestRun.step("Start cache and add core."):
+        cache = casadm.start_cache(cache_device, cache_mode, cls)
+        core = cache.add_core(core_device)
+
+    with TestRun.step("Copy file to CAS."):
+        copy_file(source=source_file.full_path, target=core.system_path,
+                  size=test_file_size, direct="oflag")
+        TestRun.LOGGER.info(str(core.get_statistics()))
+
+    with TestRun.step("Unplug cache device."):
+        cache_disk.unplug()
+        TestRun.LOGGER.info(f"List caches:\n{casadm.list_caches().stdout}")
+        TestRun.LOGGER.info(f"Dirty blocks on cache: "
+                            f"{cache.get_dirty_blocks().get_value(Unit.Blocks4096)}")
+
+    with TestRun.step("Stop cache."):
+        cache.stop()
+
+    with TestRun.step("Plug missing cache device."):
+        cache_disk.plug()
+
+    with TestRun.step("Load cache."):
+        cache = casadm.load_cache(cache_device)
+        TestRun.LOGGER.info(f"Dirty blocks on cache: "
+                            f"{cache.get_dirty_blocks().get_value(Unit.Blocks4096)}")
+
+    with TestRun.step("Stop cache with data flush."):
+        cache.stop()
+
+    with TestRun.step("Copy file from core device and check md5sum."):
+        copy_file(source=core_device.system_path, target=target_file.full_path,
+                  size=test_file_size, direct="iflag")
+        compare_files(source_file, target_file)
+
+    with TestRun.step("Cleanup core device and remove test files."):
+        Dd() \
+            .input("/dev/zero") \
+            .output(core_device.system_path) \
+            .block_size(Size(1, Unit.Blocks4096)) \
+            .count(10) \
+            .oflag("direct") \
+            .run()
```

Comment:
why is this needed?

Reply:
At the beginning I had more than one iteration in this test so I had to clean core before next iteration, but I changed it to parametrize, so it can be removed now